src/main/java/junmin/netflix/costreportservice/service/EPCostReportService.java

```diff
@@ -0,0 +1,169 @@
+			String epCode = cost.getEpisodeCode();
+			BigDecimal amount = cost.getAmount();
+			if(prodCostMap.get(epCode)==null){
+				prodCostMap.put(epCode, amount);
+			}else{
+				prodCostMap.put(epCode, prodCostMap.get(epCode).add(amount));
+			}
+//			epItemizedCostRepo.save(new EPItemizedCostEntity(prodName, epCode, amount, timestamp));
+		}
+
+		// store each EP total cost, return back EP total cost summary
+		List<EPCostRecord> totalCostRecords = new ArrayList<>();
+		for(String key : prodCostMap.keySet()){
+			epTotalCostRepo.save(new EPTotalCostEntity(prodName, key, prodCostMap.get(key), null, prodCostMap.get(key), timestamp));
+			totalCostRecords.add(new EPCostRecord(key, prodCostMap.get(key)));
+		}
+		return totalCostRecords;
+	}
+
+	public List<EPCostRecord> processAmortizedProdCostReport(String prodName, List<EPCostRecord> itemizedCosts) throws InvalidReportException
+	{
+		validateProdName(prodName);
+		// store ep keys in order in the Map
+		Map<String, BigDecimal> prodCostMap = new TreeMap<>();
+
+		// store each itemized cost into DB, and calculate total cost for each EPCode
+		long timestamp = System.currentTimeMillis();
+		for(EPCostRecord cost : itemizedCosts){
+			String epCode = cost.getEpisodeCode();
+			BigDecimal amount = cost.getAmount();
+			if(prodCostMap.get(epCode)==null){
+				prodCostMap.put(epCode, amount);
+			}else{
+				prodCostMap.put(epCode, prodCostMap.get(epCode).add(amount));
+			}
+//			epItemizedCostRepo.save(new EPItemizedCostEntity(prodName, epCode, amount, timestamp));
+		}
+
+		// calculate amortized cost
+		BigDecimal amoritzedCost = prodCostMap.get(AMORIZEDCODE);
+		BigDecimal epAmortizedCost = new BigDecimal(0);
+		if(amoritzedCost!=null) {
+			int totalEps = prodCostMap.keySet().size()-1;
+			epAmortizedCost = amoritzedCost.divide(new BigDecimal(totalEps), 2, RoundingMode.HALF_EVEN);;
+		}
+
+		// store each EP total cost including amortized cost, return back EP total cost summary
+		List<EPCostRecord> totalCostRecords = new ArrayList<>();
+		for(String key : prodCostMap.keySet()){
+			if(key.equalsIgnoreCase(AMORIZEDCODE)) continue;
+			epTotalCostRepo.save(new EPTotalCostEntity(prodName, key, prodCostMap.get(key), epAmortizedCost, prodCostMap.get(key).add(epAmortizedCost), timestamp));
+			totalCostRecords.add(new EPCostRecord(key, prodCostMap.get(key).add(epAmortizedCost)));
+		}
+		return totalCostRecords;
+	}
+
+	private String getEPCodeFromEPReport(List<EPCostRecord> costList) throws InvalidReportException {
+		// throw InvalidReportException if report is empty
+		if(costList==null || costList.size()==0) {
+			throw createInvalidReportException(ErrorEnum.EMPTY_REPORT, "cost report is empty");
+		}
+
+		String epCode = null;
+		for(EPCostRecord cost : costList){
+			if(epCode == null){
+				epCode = cost.getEpisodeCode();
+			}
+			// throw InvalidReportException if a single EP cost report contains multiple EP code,
+			if(epCode!=null && !epCode.equalsIgnoreCase(cost.getEpisodeCode())){
```

Comment:
1. Based on the requirement doc, it asked three new endpoints to process single EP cost report, production cost report and production cost report with amortized amount. So that what I followed. Imagine for process single EP cost report, if you have two EP code in the report, I would consider it to be error and give alert. 
2. If we do not require 3 new endpoints, I would just combine all three scenarios into one POST operation: production cost report with amortized amount, it's sufficient to cover all cases.